onnxruntime/core/providers/cpu/math/element_wise_ops.cc

```diff
@@ -273,25 +275,89 @@ Status Sqrt<T>::Compute(OpKernelContext* ctx) const {
   return Status::OK();
 }
 
-template <typename T>
-Status Pow<T>::Compute(OpKernelContext* context) const {
-  const Tensor& Y = *context->Input<Tensor>(1);
-  std::function<void(EigenVectorMap<T>, ConstEigenVectorMap<T>, T)> input1scalar =
-      [](EigenVectorMap<T> output, ConstEigenVectorMap<T> input0, T input1) { output = Eigen::pow(input0.array(), input1); };
-  if (Y.Shape().Size() == 1) {
-    T value = *Y.Data<T>();
-    if (value == 2.0) {
-      input1scalar = [](EigenVectorMap<T> output, ConstEigenVectorMap<T> input0, T) { output = Eigen::square(input0.array()); };
-    } else if (value == 3.0) {
-      input1scalar = [](EigenVectorMap<T> output, ConstEigenVectorMap<T> input0, T) { output = Eigen::cube(input0.array()); };
-    }
```

Comment:
Eigen::square and Eigen::cube are massively faster with floationg point values  than Eigen::pow. If that also applies to std::pow we need to keep these special cases.